idaes/power_generation/costing/tests/test_NGCC_costing.py

```diff
@@ -0,0 +1,352 @@
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Case B31A - Natural Gas Combined Cycle (NGCC) plant_gross_power
+Reference: NETL Baseline Report Rev 4
+Author: A. Deshpande and M. Zamarripa
+"""
+import pytest
+from idaes.power_generation.costing.power_plant_costing import \
+     (get_PP_costing,
+      build_flowsheet_cost_constraint,
+      costing_initialization)
+from idaes.core.util.model_statistics import degrees_of_freedom
+import pyomo.environ as pyo
+from idaes.core import FlowsheetBlock
+from idaes.core.util import get_solver
+
+# Get default solver for testing
+solver = get_solver()
+
+
+@pytest.mark.unit
+def test_get_costing():
+    # Create a Concrete Model as the top level object
+    m = pyo.ConcreteModel()
+
+    # Add a flowsheet object to the model
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.get_costing(year='2018')
+
+    # Accounts with Feedwater Flow to HP section of HRSG, as the
+    # reference/scaling parameter - Exhibit 5-15
+    FW_accounts = ['3.1', '3.3', '8.4']
+
+    m.fs.unit = pyo.Block()
+    m.fs.unit.feedwater_flowrate = pyo.Var(initialize=1085751)  # lb/hr
+    m.fs.unit.feedwater_flowrate.fix()
+    get_PP_costing(m.fs.unit, FW_accounts,
+                   m.fs.unit.feedwater_flowrate, 'lb/hr', 6)
+    assert isinstance(m.fs.unit.costing.total_plant_cost, pyo.Var)
+    assert hasattr(m.fs.unit.costing, "bare_erected_cost")
+    assert isinstance(m.fs.unit.costing.total_plant_cost_eq, pyo.Constraint)
+    assert degrees_of_freedom(m) == 0
+
+
+@pytest.mark.solver
+@pytest.mark.skipif(solver is None, reason="Solver not available")
+@pytest.mark.component
+def test_get_ngcc_costing():
+    # Create a Concrete Model as the top level object
+    m = pyo.ConcreteModel()
+
+    # Add a flowsheet object to the model
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    m.fs.get_costing(year='2018')
+
+    # Accounts corresponding to each component of the NGCC plant
+    # Reference - Exhibit 5-17, NETL Baseline Report Rev 4
+    NGCC_accounts = ['3.1', '3.2', '3.3', '3.4', '3.5', '3.6', '3.7', '3.9',
+                     '6.1', '6.3', '6.4', '6.5', '7.1', '7.2', '7.3', '7.4',
+                     '7.5', '7.6', '8.1', '8.2', '8.3', '8.4', '8.5', '9.1',
+                     '9.2', '9.3', '9.4', '9.5', '9.6', '9.7', '11.1', '11.2',
+                     '11.3', '11.4', '11.5', '11.6', '11.7', '11.8', '11.9',
+                     '12.1', '12.2', '12.3', '12.4', '12.5', '12.6', '12.7',
+                     '12.8', '12.9', '13.1', '13.2', '13.3', '14.1', '14.3',
+                     '14.4', '14.5', '14.6', '14.7', '14.8', '14.9', '14.10']
+    # Accounts with Feedwater Flow to HP section of HRSG, as the
+    # reference/scaling parameter - Exhibit 5-15
+    FW_accounts = ['3.1', '3.3', '8.4']
+    m.fs.b1 = pyo.Block()
+
+    m.fs.b1.feedwater_flowrate = pyo.Var(initialize=1085751)  # lb/hr
```

Comment:
It would be nice to see units defined here, so that you can do an unit consistency check later.

Reply:
This costing framework avoids passing units, the user must do unit conversions before passing the reference parameter.